lib/mgmcong.gi

```diff
@@ -588,8 +588,7 @@ BindGlobal("MagmaCongruencePartition",
                             ## No holes are left is at the end otherwise
                             ##    move the last one into the middle
                             if not p2=Length(forest) then
-                                forest[p2]:=forest[Length(forest)];
-                                Unbind(forest[Length(forest)]);
+                                forest[p2]:=Remove(forest);
```

Comment:
OK that's another case similar to the ones @ChrisJefferson pointed out as problematic, and it causes a test failure. I'll now look through the PR for all similar instances and revert/adapt them, and will say once this is done (then perhaps @ChrisJefferson can double check ... ;-))

Reply:
Although... wait, this code is not the same, as it explicitly checks that `not p2=Length(forest)` which should be the same as `p2 <> Length(forest)`... 

Alas, in the failing case, it seems we actually have `p2 > Length(forest)` ... So the original code sets `forest[p2]` and then removes that... So I think we can adjust that check to `p2 < Length(forest)`